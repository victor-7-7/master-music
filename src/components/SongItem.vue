<template>
  <!--Вариант с классами (section 20, video 012, time 5:00)-->
  <!--https://michaelnthiessen.com/dynamically-add-class-name-->
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50" :class="`song-id-${song.docId}`" :id="`song-id-${song.docId}`"
  >
    <div>
      <router-link
        :to="{ name: 'song', params: { id: song.docId } }"
        class="font-bold block text-gray-600"
      >
        {{ song.modified_name }}
      </router-link>
      <!-- имя юзера, который выгрузил песню в хранилище -->
      <span class="text-gray-500 text-sm">{{ song.user_display_name }}</span>
    </div>

    <div class="text-gray-600 text-lg">
      <!-- custom attr дает нам контроль над поведением при навигации,
       мы хотим при переходе на страницу song проскроллить к фрагменту comments-->
      <router-link custom
           :to="{ name: 'song', params: { id: song.docId }, hash: '#comments'}"
           v-slot="{ navigate }">
        <span class="comments" @click="navigate">
          <i class="fa fa-comments text-gray-600"></i>
          {{ song.comment_count }}
        </span>
      </router-link>
    </div>
  </li>
</template>

<script>
export default {
  name: 'SongItem',
  props: {
    // Свойство song будет проброшено сюда из родительского компонента Home.vue
    // через v-bind атрибут - :song
    song: {
      type: Object,
      // Свойство song является обязательным. Если song в родительском
      // компоненте будет undefined, то движок кинет ворнинг
      required: true,
    },
  },

}
</script>

<style scoped>

</style>