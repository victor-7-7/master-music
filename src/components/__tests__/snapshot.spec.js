// https://vitest.dev/guide/snapshot.html
// В снэпшоте компонента мы сохраняем его рендер (что компонент
// предоставит браузеру - html fragment), а в тесте сравниваем
// текущий рендер компонента с сохраненным в снэпшоте

// import { describe, it, expect } from 'vitest' // <-- не требуется!

import SongItem from '@/components/SongItem.vue'
import { RouterLinkStub, shallowMount } from '@vue/test-utils'

describe('Snapshot SongItem.vue', () => {
  it('renders correctly', () => {
    const song = {
      docId: "abc",
      modified_name: "test",
      user_display_name: "test",
      comment_count: 5,
    }

    const wrapper = shallowMount(SongItem, {
      props: { song, },
      global: {
        components: {
          'router-link': RouterLinkStub,
        }
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})