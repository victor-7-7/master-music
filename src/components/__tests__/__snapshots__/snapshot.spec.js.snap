// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot SongItem.vue > renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!--Вариант с классами (section 20, video 012, time 5:00)-->
  <!--https://michaelnthiessen.com/dynamically-add-class-name-->
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 song-id-abc"
    id="song-id-abc"
  >
    <div>
      <router-link-stub
        class="font-bold block text-gray-600"
        custom="false"
        to="[object Object]"
      />
      <!-- имя юзера, который выгрузил песню в хранилище -->
      <span
        class="text-gray-500 text-sm"
      >
        test
      </span>
    </div>
    <div
      class="text-gray-600 text-lg"
    >
      <!-- custom attr дает нам контроль над поведением при навигации,
       мы хотим при переходе на страницу song проскроллить к фрагменту comments-->
      <router-link-stub
        custom="true"
        to="[object Object]"
      />
    </div>
  </li>
  
</div>
`;
